pandas/tests/io/test_excel.py

```diff
@@ -64,136 +63,93 @@ def setup_method(self, datapath):
 class ReadingTestsBase(SharedItems):
     # This is based on ExcelWriterBase
 
+    @pytest.fixture(autouse=True)
+    def change_to_data_directory(self, datapath):
+        cwd = os.getcwd()
+        os.chdir(datapath("io", "data"))
+        yield
+        os.chdir(cwd)
+
     @pytest.fixture(autouse=True, params=['xlrd', None])
     def set_engine(self, request):
-        func_name = "get_exceldf"
-        old_func = getattr(self, func_name)
-        new_func = partial(old_func, engine=request.param)
-        setattr(self, func_name, new_func)
+        global read_excel
+        original_func = read_excel
+        read_excel = partial(read_excel, engine=request.param)
         yield
-        setattr(self, func_name, old_func)
+        read_excel = original_func
```

Comment:
using monkeypatch in a fixture would negate the need for this boilerplate

Reply:
Great idea actually - let me take a look at that